obs_encoder.py

```diff
@@ -0,0 +1,266 @@
+import treetensor.torch as ttorch
```

Comment:
add unittest for file, especially taking loss back propogration into consideration